storybookjs/storybook
# SvelteKit: Add experimental page and navigation mocking

Depends on https://github.com/storybookjs/storybook/pull/24921

Closes #20999

## What I did

This PR adds a tighter integration for sveltekit allowing the user to mock the stores from sveltekit, adds an override listener for the links allowing you to execute code when a link it's clicked intead of navigating, adding the ability to still import and run code on `afterNavigate` (here emulated like `onMount`), adding callbacks for the other functions like `goto`, `invalidateAll`.

For the moment is a draft PR. I'm getting some weird compilation errors whenever i try to dynamically import `@storybook/addon-actions` but overall it should already work kinda fine.

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

1. Run a sandbox for template sveltekit
2. Modify the Button component to `import { page } from "$app/stores"`
3. print the content of the store on the page with `{JSON.stringify($page)}`
4. verify that the component normally work when sveltekit is launched
5. Modify `Button.stories.ts` and add to the primary button `parameters.sveltekit.stores.page={ test: true }`
6. Open Storybook in your browser
7. Access the Button story
8. verify that now the stringified page store contains `{ "test": true }`


### Documentation




## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
